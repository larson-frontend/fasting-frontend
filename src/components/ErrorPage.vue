<template>
  <div class="min-h-screen bg-gray-50 flex items-center justify-center p-4">
    <div class="max-w-md w-full">
      <!-- Error Icon -->
      <div class="text-center mb-8">
        <div class="w-20 h-20 mx-auto mb-4 rounded-full bg-red-100 flex items-center justify-center">
          <svg class="w-10 h-10 text-red-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" 
                  d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.732-.833-2.5 0L4.314 16.5c-.77.833.192 2.5 1.732 2.5z">
            </path>
          </svg>
        </div>
        <h1 class="text-2xl font-bold text-gray-900 mb-2">{{ $t('errors.connection_error') }}</h1>
        <p class="text-gray-600">
          {{ errorMessage }}
        </p>
      </div>

      <!-- Error Details -->
      <div class="bg-white rounded-lg border border-red-200 p-6 mb-6">
        <div class="space-y-4">
          <!-- Status -->
          <div class="flex items-center justify-between">
            <span class="text-sm font-medium text-gray-700">{{ $t('errors.status') }}:</span>
            <span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-red-100 text-red-800">
              {{ isOnline ? $t('errors.service_unavailable') : $t('errors.offline') }}
            </span>
          </div>

          <!-- Last Attempt -->
          <div class="flex items-center justify-between">
            <span class="text-sm font-medium text-gray-700">{{ $t('errors.last_attempt') }}:</span>
            <span class="text-sm text-gray-600">{{ lastAttempt }}</span>
          </div>

          <!-- Error Code -->
          <div v-if="errorCode" class="flex items-center justify-between">
            <span class="text-sm font-medium text-gray-700">{{ $t('errors.error_code') }}:</span>
            <span class="text-sm font-mono text-gray-600">{{ errorCode }}</span>
          </div>
        </div>
      </div>

      <!-- Action Buttons -->
      <div class="space-y-3">
        <!-- Retry Button -->
        <button
          @click="retry"
          :disabled="isRetrying"
          class="w-full flex items-center justify-center px-4 py-3 border border-transparent rounded-lg shadow-sm text-sm font-medium text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 disabled:opacity-50 disabled:cursor-not-allowed transition-colors"
        >
          <svg v-if="isRetrying" class="animate-spin -ml-1 mr-3 h-5 w-5 text-white" fill="none" viewBox="0 0 24 24">
            <circle class="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4"></circle>
            <path class="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"></path>
          </svg>
          {{ isRetrying ? $t('errors.retrying') : $t('errors.retry') }}
        </button>

        <!-- Refresh Page -->
        <button
          @click="refreshPage"
          class="w-full flex items-center justify-center px-4 py-3 border border-gray-300 rounded-lg shadow-sm text-sm font-medium text-gray-700 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 transition-colors"
        >
          <svg class="w-4 h-4 mr-2" fill="none" stroke="currentColor" viewBox="0 0 24 24">
            <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15"></path>
          </svg>
          {{ $t('errors.refresh_page') }}
        </button>
      </div>

      <!-- Help Section -->
      <div class="mt-8 text-center">
        <div class="text-sm text-gray-600 space-y-2">
          <p><strong>{{ $t('errors.possible_causes') }}:</strong></p>
          <ul class="text-left space-y-1 text-xs">
            <li>• {{ $t('errors.no_internet') }}</li>
            <li>• {{ $t('errors.server_unavailable') }}</li>
            <li>• {{ $t('errors.maintenance') }}</li>
            <li>• {{ $t('errors.firewall_blocked') }}</li>
          </ul>
        </div>
        
        <!-- Contact Support -->
        <div class="mt-6 pt-6 border-t border-gray-200">
          <p class="text-xs text-gray-500 mb-2">
            {{ $t('errors.problem_persists') }}
          </p>
          <a 
            href="mailto:support@fastingtracker.com" 
            class="text-xs text-blue-600 hover:text-blue-500 underline"
          >
            {{ $t('errors.contact_support') }}
          </a>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref, computed, onMounted, onUnmounted } from 'vue'
import { useI18n } from 'vue-i18n'

interface Props {
  error?: Error
  retryCallback?: () => Promise<void>
}

const props = defineProps<Props>()
const emit = defineEmits<{
  retry: []
  close: []
}>()

const { t } = useI18n()
const isRetrying = ref(false)
const lastAttempt = ref(new Date().toLocaleTimeString())
const onlineStatus = ref(navigator.onLine)

// Computed Properties
const errorMessage = computed(() => {
  if (!onlineStatus.value) {
    return t('errors.no_connection_message')
  }
  
  if (props.error?.message.includes('NetworkError') || props.error?.message.includes('fetch')) {
    return t('errors.service_unreachable_message')
  }
  
  return t('errors.unexpected_error')
})

const isOnline = computed(() => onlineStatus.value)

const errorCode = computed(() => {
  if (props.error) {
    // Extract error code from message or response
    if (props.error.message.includes('500')) return 'HTTP 500'
    if (props.error.message.includes('404')) return 'HTTP 404'
    if (props.error.message.includes('timeout')) return 'TIMEOUT'
    if (props.error.message.includes('NetworkError')) return 'NETWORK_ERROR'
    return 'UNKNOWN_ERROR'
  }
  return null
})

// Methods
async function retry() {
  if (isRetrying.value) return
  
  isRetrying.value = true
  lastAttempt.value = new Date().toLocaleTimeString()
  
  try {
    if (props.retryCallback) {
      await props.retryCallback()
      emit('close') // Schließe Error-Seite bei Erfolg
    } else {
      emit('retry')
    }
  } catch (error) {
    console.error('Retry failed:', error)
    // Bleibe auf Error-Seite
  } finally {
    isRetrying.value = false
  }
}

function refreshPage() {
  window.location.reload()
}

// Online/Offline Detection
function updateOnlineStatus() {
  onlineStatus.value = navigator.onLine
}

// Lifecycle
onMounted(() => {
  window.addEventListener('online', updateOnlineStatus)
  window.addEventListener('offline', updateOnlineStatus)
})

onUnmounted(() => {
  window.removeEventListener('online', updateOnlineStatus)
  window.removeEventListener('offline', updateOnlineStatus)
})
</script>

<style scoped>
.animate-spin {
  animation: spin 1s linear infinite;
}

@keyframes spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
</style>
