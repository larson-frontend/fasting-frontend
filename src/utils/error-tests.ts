/**
 * Manual Error Testing
 * Praktische Tests f√ºr verschiedene Error-Szenarien
 * 
 * Diese Tests k√∂nnen manuell √ºber das TestPanel oder Browser-Console ausgef√ºhrt werden
 */

// Browser Console Commands f√ºr Error Testing
if (typeof window !== 'undefined') {
  const errorTests = {
    
    /**
     * 1. Network Error Tests
     */
    simulateNetworkError: () => {
      console.log('üîå Simuliere Netzwerk-Fehler...')
      
      // Mock fetch to fail
      const originalFetch = window.fetch
      window.fetch = () => Promise.reject(new Error('NetworkError: Failed to fetch'))
      
      console.log('‚úÖ Network Error simuliert - versuche API-Call √ºber TestPanel')
      console.log('üîÑ Zum Zur√ºcksetzen: errorTests.resetFetch()')
      
      return { originalFetch }
    },

    /**
     * 2. API Server Error Tests  
     */
    simulateServerError: (statusCode = 500) => {
      console.log(`üö® Simuliere Server Error ${statusCode}...`)
      
      const originalFetch = window.fetch
      window.fetch = () => Promise.resolve({
        ok: false,
        status: statusCode,
        statusText: statusCode === 500 ? 'Internal Server Error' : 'Not Found',
        json: () => Promise.resolve({ error: `Server Error ${statusCode}` })
      } as Response)
      
      console.log(`‚úÖ Server Error ${statusCode} simuliert`)
      return { originalFetch }
    },

    /**
     * 3. Timeout Error Tests
     */
    simulateTimeout: (delay = 10000) => {
      console.log(`‚è±Ô∏è Simuliere Timeout nach ${delay}ms...`)
      
      const originalFetch = window.fetch
      window.fetch = () => new Promise((_, reject) => 
        setTimeout(() => reject(new Error(`Request timeout after ${delay}ms`)), delay)
      )
      
      console.log('‚úÖ Timeout simuliert')
      return { originalFetch }
    },

    /**
     * 4. CORS Error Tests
     */
    simulateCORSError: () => {
      console.log('üö´ Simuliere CORS-Fehler...')
      
      const originalFetch = window.fetch
      window.fetch = () => Promise.reject(
        new Error('CORS policy: No Access-Control-Allow-Origin header is present on the requested resource')
      )
      
      console.log('‚úÖ CORS Error simuliert')
      return { originalFetch }
    },

    /**
     * 5. Malformed JSON Response
     */
    simulateMalformedJSON: () => {
      console.log('üìÑ Simuliere defekte JSON-Response...')
      
      const originalFetch = window.fetch
      window.fetch = () => Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.reject(new Error('Unexpected token < in JSON at position 0'))
      } as Response)
      
      console.log('‚úÖ Malformed JSON simuliert')
      return { originalFetch }
    },

    /**
     * 6. Offline Simulation
     */
    simulateOffline: () => {
      console.log('üì± Simuliere Offline-Modus...')
      
      // Override navigator.onLine
      Object.defineProperty(navigator, 'onLine', {
        writable: true,
        value: false
      })
      
      // Dispatch offline event
      window.dispatchEvent(new Event('offline'))
      
      console.log('‚úÖ Offline-Modus simuliert')
      console.log('üîÑ Zum Zur√ºcksetzen: errorTests.simulateOnline()')
    },

    /**
     * 7. Online Simulation (Reset)
     */
    simulateOnline: () => {
      console.log('üåê Simuliere Online-Modus...')
      
      Object.defineProperty(navigator, 'onLine', {
        writable: true,
        value: true
      })
      
      window.dispatchEvent(new Event('online'))
      
      console.log('‚úÖ Online-Modus wiederhergestellt')
    },

    /**
     * Reset Fetch (f√ºr alle Network Mocks)
     */
    resetFetch: () => {
      console.log('üîÑ Setze fetch zur√ºck...')
      
      // Restore original fetch
      if ('originalFetch' in window) {
        window.fetch = (window as any).originalFetch
        delete (window as any).originalFetch
      }
      
      console.log('‚úÖ Fetch zur√ºckgesetzt')
    },

    /**
     * Production Error Page Test
     */
    triggerProductionError: () => {
      console.log('üö® Teste Production Error Page...')
      console.log('‚ö†Ô∏è  Achtung: Funktioniert nur in Production Build!')
      
      // Simulate error in production
      errorTests.simulateNetworkError()
      
      // Try to trigger API call
      console.log('üîÑ Rufe API auf um Error-Seite zu triggern...')
      
      // You would call your API here
      console.log('üìù Manueller Schritt: Klicke auf Start/Stop Button in der App')
    },

    /**
     * Development Fallback Test
     */
    testDevelopmentFallback: () => {
      console.log('üîß Teste Development Fallback...')
      console.log('‚ö†Ô∏è  Funktioniert nur in Development Mode!')
      
      errorTests.simulateNetworkError()
      
      console.log('üìù Manueller Schritt: Verwende TestPanel f√ºr API-Calls')
      console.log('üîç Erwarte: Toast-Benachrichtigung und Fallback zu Mock-Daten')
    },

    /**
     * Alle Tests zur√ºcksetzen
     */
    resetAll: () => {
      console.log('üîÑ Setze alle Tests zur√ºck...')
      
      errorTests.resetFetch()
      errorTests.simulateOnline()
      
      console.log('‚úÖ Alle Tests zur√ºckgesetzt')
    },

    /**
     * Test-Suite ausf√ºhren
     */
    runTestSuite: async () => {
      console.log('üß™ Starte Error Test Suite...')
      console.log('')
      
      const tests = [
        { name: 'Network Error', fn: errorTests.simulateNetworkError },
        { name: 'Server Error 500', fn: () => errorTests.simulateServerError(500) },
        { name: 'Server Error 404', fn: () => errorTests.simulateServerError(404) },
        { name: 'Timeout', fn: () => errorTests.simulateTimeout(3000) },
        { name: 'CORS Error', fn: errorTests.simulateCORSError },
        { name: 'Malformed JSON', fn: errorTests.simulateMalformedJSON },
        { name: 'Offline', fn: errorTests.simulateOffline }
      ]
      
      for (const test of tests) {
        console.log(`\nüîç Test: ${test.name}`)
        test.fn()
        console.log('üìù Teste jetzt die App-Funktionalit√§t...')
        
        // Wait for manual testing
        await new Promise(resolve => {
          console.log('‚è∏Ô∏è  Dr√ºcke Enter zum n√§chsten Test...')
          // In real scenario, you'd wait for user input
          setTimeout(resolve, 2000) // Auto-continue for demo
        })
        
        errorTests.resetAll()
      }
      
      console.log('\n‚úÖ Test Suite abgeschlossen!')
    },

    /**
     * Help - Zeige alle verf√ºgbaren Tests
     */
    help: () => {
      console.log('üß™ Verf√ºgbare Error Tests:')
      console.log('')
      console.log('=== Netzwerk Tests ===')
      console.log('‚Ä¢ errorTests.simulateNetworkError() - Netzwerk-Fehler')
      console.log('‚Ä¢ errorTests.simulateTimeout() - Request Timeout')
      console.log('‚Ä¢ errorTests.simulateCORSError() - CORS-Fehler')
      console.log('‚Ä¢ errorTests.simulateOffline() - Offline-Modus')
      console.log('‚Ä¢ errorTests.simulateOnline() - Online-Modus')
      console.log('')
      console.log('=== API Tests ===')
      console.log('‚Ä¢ errorTests.simulateServerError(500) - Server Error')
      console.log('‚Ä¢ errorTests.simulateServerError(404) - Not Found')
      console.log('‚Ä¢ errorTests.simulateMalformedJSON() - Defekte JSON Response')
      console.log('')
      console.log('=== Spezial Tests ===')
      console.log('‚Ä¢ errorTests.triggerProductionError() - Production Error Page')
      console.log('‚Ä¢ errorTests.testDevelopmentFallback() - Development Fallback')
      console.log('‚Ä¢ errorTests.runTestSuite() - Alle Tests automatisch')
      console.log('')
      console.log('=== Reset ===')
      console.log('‚Ä¢ errorTests.resetFetch() - Fetch zur√ºcksetzen')
      console.log('‚Ä¢ errorTests.resetAll() - Alles zur√ºcksetzen')
      console.log('')
      console.log('üìù Nach jedem Test: Verwende TestPanel oder App-Buttons zum Testen')
    }
  }

  // Expose to window
  ;(window as any).errorTests = errorTests

  // Log availability
  console.log('üß™ Error Tests verf√ºgbar unter: window.errorTests')
  console.log('üí° Hilfe: errorTests.help()')
}

export {};
