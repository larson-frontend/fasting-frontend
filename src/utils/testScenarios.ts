/**
 * Test-Szenarien für Fasting Tracker
 * 
 * Diese Datei enthält verschiedene Test-Szenarien die beim Entwickeln 
 * und Testen der Fasten-App verwendet werden können.
 * 
 * Verwendung:
 * 1. Importiere die gewünschte Test-Funktion
 * 2. Rufe sie in der Komponente auf
 * 3. Die Test-Daten überschreiben temporär die echten Props
 */

export interface TestScenario {
  hours: number
  minutes: number
  goalHours?: number
  description: string
  expectedPhase: string
  expectedColor: string
}

// Vordefinierte Test-Szenarien
export const TEST_SCENARIOS: TestScenario[] = [
  {
    hours: 0,
    minutes: 30,
    goalHours: 16,
    description: "Start Phase - 30 Minuten (Ziel: 16h)",
    expectedPhase: "Aufwärmphase",
    expectedColor: "orange"
  },
  {
    hours: 8,
    minutes: 15,
    goalHours: 12,
    description: "Fettverbrennung - 8h 15min (Ziel: 12h)",
    expectedPhase: "Fettverbrennung",
    expectedColor: "orange"
  },
  {
    hours: 11,
    minutes: 0,
    goalHours: 10,
    description: "10h Ziel überschritten - 11h (Ziel: 10h)",
    expectedPhase: "Bonus-Zeit",
    expectedColor: "purple"
  },
  {
    hours: 13,
    minutes: 30,
    goalHours: 12,
    description: "12h Ziel überschritten - 13h 30min (Ziel: 12h)",
    expectedPhase: "Bonus-Zeit",
    expectedColor: "purple"
  },
  {
    hours: 16,
    minutes: 0,
    goalHours: 16,
    description: "16h Ziel erreicht - 16h exakt (Ziel: 16h)",
    expectedPhase: "16h Ziel erreicht!",
    expectedColor: "green"
  },
  {
    hours: 17,
    minutes: 30,
    goalHours: 16,
    description: "Über 16h Ziel - 17h 30min (Ziel: 16h)",
    expectedPhase: "Bonus-Zeit",
    expectedColor: "purple"
  },
  {
    hours: 19,
    minutes: 15,
    goalHours: 18,
    description: "18h Ziel überschritten - 19h 15min (Ziel: 18h)",
    expectedPhase: "Bonus-Zeit",
    expectedColor: "purple"
  },
  {
    hours: 24,
    minutes: 0,
    goalHours: 24,
    description: "24h Ziel erreicht - 24h (Ziel: 24h)",
    expectedPhase: "24h Ziel erreicht!",
    expectedColor: "green"
  }
]

// Aktueller Test-Modus
let currentTestMode = false
let currentTestData: TestScenario | null = null

/**
 * Aktiviert einen Test-Modus mit spezifischen Stunden/Minuten
 */
export function activateTestMode(hours: number, minutes: number = 0, goalHours: number = 16): void {
  currentTestMode = true
  currentTestData = {
    hours,
    minutes,
    goalHours,
    description: `Test: ${hours}h ${minutes}m (Ziel: ${goalHours}h)`,
    expectedPhase: getExpectedPhase(hours, goalHours),
    expectedColor: getExpectedColor(hours, goalHours)
  }
  
  console.log(`🧪 Test-Modus aktiviert: ${hours}h ${minutes}m (Ziel: ${goalHours}h)`)
  console.log(`📊 Erwartete Phase: ${currentTestData.expectedPhase}`)
  console.log(`🎨 Erwartete Farbe: ${currentTestData.expectedColor}`)
}

/**
 * Aktiviert ein vordefiniertes Test-Szenario
 */
export function activateTestScenario(scenarioName: string): void {
  const scenario = TEST_SCENARIOS.find(s => s.description.includes(scenarioName))
  if (scenario) {
    currentTestMode = true
    currentTestData = scenario
    console.log(`🧪 Test-Szenario aktiviert: ${scenario.description}`)
    console.log(`📊 Erwartete Phase: ${scenario.expectedPhase}`)
    console.log(`🎨 Erwartete Farbe: ${scenario.expectedColor}`)
  } else {
    console.error(`❌ Test-Szenario '${scenarioName}' nicht gefunden`)
  }
}

/**
 * Deaktiviert den Test-Modus (zurück zu echten Daten)
 */
export function deactivateTestMode(): void {
  currentTestMode = false
  currentTestData = null
  console.log(`✅ Test-Modus deaktiviert - verwende echte Daten`)
}

/**
 * Gibt die aktuellen Test-Daten zurück (falls Test-Modus aktiv)
 */
export function getTestData(): { hours: number; minutes: number; goalHours?: number } | null {
  if (currentTestMode && currentTestData) {
    return {
      hours: currentTestData.hours,
      minutes: currentTestData.minutes,
      goalHours: currentTestData.goalHours
    }
  }
  return null
}

/**
 * Prüft ob Test-Modus aktiv ist
 */
export function isTestModeActive(): boolean {
  return currentTestMode
}

/**
 * Zeigt alle verfügbaren Test-Szenarien in der Konsole
 */
export function listTestScenarios(): void {
  console.log("🧪 Verfügbare Test-Szenarien:")
  TEST_SCENARIOS.forEach((scenario, index) => {
    console.log(`${index + 1}. ${scenario.description} (${scenario.hours}h ${scenario.minutes}m)`)
  })
}

// Hilfsfunktionen
function getExpectedPhase(hours: number, goalHours: number = 16): string {
  if (hours < 3) return "Aufwärmphase"
  if (hours < 8) return "Aufwärmphase"
  if (hours < 12) return "Fettverbrennung"
  if (hours < goalHours) return "Ketose"
  if (hours === goalHours) return `${goalHours}h Ziel erreicht!`
  return "Bonus-Zeit"
}

function getExpectedColor(hours: number, goalHours: number = 16): string {
  if (hours < goalHours) return "orange"
  if (hours === goalHours) return "green"
  return "purple"
}

// Quick-Access Funktionen für häufige Tests
export const quickTests = {
  start: () => activateTestMode(0, 30, 16),
  early: () => activateTestMode(3, 0, 16),
  middle: () => activateTestMode(8, 15, 16),
  ketosis: () => activateTestMode(12, 30, 16),
  autophagy: () => activateTestMode(16, 0, 16),
  extended: () => activateTestMode(17, 30, 16),
  deep: () => activateTestMode(20, 45, 24),
  full: () => activateTestMode(24, 0, 24),
  // Test verschiedene Ziele
  goal10: () => activateTestMode(11, 0, 10), // Über 10h Ziel
  goal12: () => activateTestMode(13, 30, 12), // Über 12h Ziel
  goal18: () => activateTestMode(19, 15, 18), // Über 18h Ziel
  off: () => deactivateTestMode()
}

// Browser Console Integration (für einfache Nutzung in DevTools)
if (typeof globalThis.window !== 'undefined' && globalThis.window === globalThis) {
  (window as any).fastingTests = {
    ...quickTests,
    activate: activateTestMode,
    scenario: activateTestScenario,
    list: listTestScenarios,
    current: () => currentTestData
  }
  console.log("🧪 Fasting Tests verfügbar unter: window.fastingTests")
}
