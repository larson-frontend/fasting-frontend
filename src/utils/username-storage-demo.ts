/**
 * Demo: How to use the logged username localStorage functionality
 * 
 * This file demonstrates how to use the new getLoggedUsername() function
 * that stores and retrieves the unique username from localStorage.
 */

import { getLoggedUsername } from '../api'

/**
 * Example: Get the currently logged username
 */
export function getCurrentLoggedUsername(): string | null {
  const username = getLoggedUsername()
  
  if (username) {
    console.log('Currently logged user:', username)
    return username
  } else {
    console.log('No user is currently logged in')
    return null
  }
}

/**
 * Example: Check if a specific user is logged in
 */
export function isUserLoggedIn(expectedUsername: string): boolean {
  const currentUsername = getLoggedUsername()
  return currentUsername === expectedUsername
}

/**
 * Example: Get a display name for the logged user
 */
export function getLoggedUserDisplayName(): string {
  const username = getLoggedUsername()
  
  if (!username) {
    return 'Guest'
  }
  
  // Capitalize first letter
  return username.charAt(0).toUpperCase() + username.slice(1)
}

/**
 * Example: Use in component logic
 */
export function exampleUsageInComponent() {
  // This is how you would use it in a Vue component:
  
  /*
  <script setup lang="ts">
  import { getLoggedUsername } from '../api'
  import { ref, onMounted } from 'vue'
  
  const currentUser = ref<string | null>(null)
  
  onMounted(() => {
    currentUser.value = getLoggedUsername()
    
    if (currentUser.value) {
      console.log('Welcome back,', currentUser.value)
    }
  })
  </script>
  
  <template>
    <div v-if="currentUser">
      Welcome back, {{ currentUser }}!
    </div>
    <div v-else>
      Please log in
    </div>
  </template>
  */
}
