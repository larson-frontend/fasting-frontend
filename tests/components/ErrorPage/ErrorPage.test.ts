import { describe, it, expect, vi } from 'vitest'
import ErrorPage from '../../../src/components/ErrorPage.vue'
import { mountWithI18n } from '../../utils/test-utils'

const mountPage = (props?: Partial<{ error?: Error; retryCallback?: () => Promise<void> }>) =>
  mountWithI18n(ErrorPage, { props: { ...(props || {}) } })

describe('ErrorPage', () => {
  it('renders with undefined error and shows generic message', () => {
    const wrapper = mountPage()
    expect(wrapper.text()).toContain('Verbindungsfehler') // errors.connection_error
    // With online = true by default in setup, expect unexpected_error
    expect(wrapper.text()).toContain('Ein unerwarteter Fehler ist aufgetreten')
    // Buttons visible
    const buttons = wrapper.findAll('button')
    expect(buttons.length).toBe(2)
  })

  it('renders network error details and code when error mentions NetworkError', () => {
    const wrapper = mountPage({ error: new Error('NetworkError: failed to fetch') })
    expect(wrapper.text()).toContain('Service nicht erreichbar') // status badge text could vary with online/offline, assert code
    expect(wrapper.text()).toContain('NETWORK_ERROR')
  })

  it('emits retry when no retryCallback is provided', async () => {
    const wrapper = mountPage()
    const retryBtn = wrapper.findAll('button')[0]
    await retryBtn.trigger('click')
    expect(wrapper.emitted('retry')).toBeTruthy()
  })

  it('calls retryCallback and emits close on success', async () => {
    const cb = vi.fn(async () => Promise.resolve())
    const wrapper = mountPage({ retryCallback: cb })
    const retryBtn = wrapper.findAll('button')[0]
    await retryBtn.trigger('click')
    expect(cb).toHaveBeenCalledTimes(1)
    expect(wrapper.emitted('close')).toBeTruthy()
  })

  it('does not emit close when retryCallback throws', async () => {
    const cb = vi.fn(async () => Promise.reject(new Error('boom')))
    const wrapper = mountPage({ retryCallback: cb })
    const retryBtn = wrapper.findAll('button')[0]
    await retryBtn.trigger('click')
    expect(cb).toHaveBeenCalledTimes(1)
    expect(wrapper.emitted('close')).toBeFalsy()
  })

  it('invokes window.location.reload on Refresh Page click', async () => {
    const reloadSpy = vi.spyOn(window.location, 'reload')
    const wrapper = mountPage()
    const refreshBtn = wrapper.findAll('button')[1]
    await refreshBtn.trigger('click')
    expect(reloadSpy).toHaveBeenCalled()
  })

  it('matches snapshot', () => {
    const wrapper = mountPage()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
