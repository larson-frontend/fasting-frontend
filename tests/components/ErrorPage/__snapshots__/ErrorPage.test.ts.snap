// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > matches snapshot 1`] = `
"<div data-v-f0df1a7e="" class="min-h-screen bg-gray-50 flex items-center justify-center p-4">
  <div data-v-f0df1a7e="" class="max-w-md w-full">
    <!-- Error Icon -->
    <div data-v-f0df1a7e="" class="text-center mb-8">
      <div data-v-f0df1a7e="" class="w-20 h-20 mx-auto mb-4 rounded-full bg-red-100 flex items-center justify-center"><svg data-v-f0df1a7e="" class="w-10 h-10 text-red-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
          <path data-v-f0df1a7e="" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.732-.833-2.5 0L4.314 16.5c-.77.833.192 2.5 1.732 2.5z"></path>
        </svg></div>
      <h1 data-v-f0df1a7e="" class="text-2xl font-bold text-gray-900 mb-2">Verbindungsfehler</h1>
      <p data-v-f0df1a7e="" class="text-gray-600">Ein unerwarteter Fehler ist aufgetreten. Bitte versuchen Sie es erneut.</p>
    </div><!-- Error Details -->
    <div data-v-f0df1a7e="" class="bg-white rounded-lg border border-red-200 p-6 mb-6">
      <div data-v-f0df1a7e="" class="space-y-4">
        <!-- Status -->
        <div data-v-f0df1a7e="" class="flex items-center justify-between"><span data-v-f0df1a7e="" class="text-sm font-medium text-gray-700">Status:</span><span data-v-f0df1a7e="" class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-red-100 text-red-800">Service nicht erreichbar</span></div><!-- Last Attempt -->
        <div data-v-f0df1a7e="" class="flex items-center justify-between"><span data-v-f0df1a7e="" class="text-sm font-medium text-gray-700">Letzter Versuch:</span><span data-v-f0df1a7e="" class="text-sm text-gray-600">20:29:05</span></div><!-- Error Code -->
        <!--v-if-->
      </div>
    </div><!-- Action Buttons -->
    <div data-v-f0df1a7e="" class="space-y-3">
      <!-- Retry Button --><button data-v-f0df1a7e="" class="w-full flex items-center justify-center px-4 py-3 border border-transparent rounded-lg shadow-sm text-sm font-medium text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 disabled:opacity-50 disabled:cursor-not-allowed transition-colors">
        <!--v-if--> Erneut versuchen
      </button><!-- Refresh Page --><button data-v-f0df1a7e="" class="w-full flex items-center justify-center px-4 py-3 border border-gray-300 rounded-lg shadow-sm text-sm font-medium text-gray-700 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 transition-colors"><svg data-v-f0df1a7e="" class="w-4 h-4 mr-2" fill="none" stroke="currentColor" viewBox="0 0 24 24">
          <path data-v-f0df1a7e="" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15"></path>
        </svg> Seite neu laden</button>
    </div><!-- Help Section -->
    <div data-v-f0df1a7e="" class="mt-8 text-center">
      <div data-v-f0df1a7e="" class="text-sm text-gray-600 space-y-2">
        <p data-v-f0df1a7e=""><strong data-v-f0df1a7e="">Mögliche Ursachen:</strong></p>
        <ul data-v-f0df1a7e="" class="text-left space-y-1 text-xs">
          <li data-v-f0df1a7e="">• Keine Internetverbindung</li>
          <li data-v-f0df1a7e="">• Server ist temporär nicht verfügbar</li>
          <li data-v-f0df1a7e="">• Wartungsarbeiten laufen</li>
          <li data-v-f0df1a7e="">• Firewall blockiert die Verbindung</li>
        </ul>
      </div><!-- Contact Support -->
      <div data-v-f0df1a7e="" class="mt-6 pt-6 border-t border-gray-200">
        <p data-v-f0df1a7e="" class="text-xs text-gray-500 mb-2">Problem besteht weiterhin?</p><a data-v-f0df1a7e="" href="mailto:support@fastingtracker.com" class="text-xs text-blue-600 hover:text-blue-500 underline">Support kontaktieren</a>
      </div>
    </div>
  </div>
</div>"
`;
